roachtest: spruce up online restore roachtests

msbutler opened the pull request:
This patch modifies the OR roachtest driver in the following ways:
- it stops the cockroach process from running on the workload node
- it removes some copy pasta from the conventional restore roachtest that
  currently does not play nice with a roachprod cluster with a workload node.
- it adds an 8tb roachtest
- it changes the naming convention of the roachtests to
  `restore/{online,offline}/workload={true,false}/<workload/scale>`
- For each online restore roachtest `(prefix restore/online/*)`, the driver now
  creates a corresponding roachtest that runs a conventional restore over the
same cluster topology and workload in order to measure post restore query
latency relative to online restore `(prefix restore/control/*)`. Further, the
driver creates an additional roachtest variant where `/workload={true,false}`
determines if the driver will run a foreground workload after the restore
completes.

Epic: none

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/114454)
<!-- Reviewable:end -->

msbutler committed:
roachtest: spruce up online restore roachtests

    This patch modifies the OR roachtest driver in the following ways:
    - it stops the cockroach process from running on the workload node
    - it removes some copy pasta from the conventional restore roachtest that
      currently does not play nice with a roachprod cluster with a workload node.
    - it adds an 8tb roachtest
    - it changes the naming convention of the roachtests to
      `restore/{online,offline}/workload={true,false}/<workload/scale>`
    - For each online restore roachtest (prefix restore/online/*), the driver now
      creates a corresponding roachtest that runs a conventional restore over the
    same cluster topology and workload in order to measure post restore query
    latency relative to online restore (prefix restore/control/*). Further, the
    driver creates an additional roachtest variant where `/workload={true,false}`
    determines if the driver will run a foreground workload after the restore
    completes.

    Epic: none

    Release note: none

msbutler commented:
TFTR!

bors r=dt
